src/python/pants/rules/core/strip_source_roots_test.py

```diff
@@ -86,7 +97,54 @@ def get_stripped_files_for_snapshot(
             get_stripped_files_for_snapshot(file_names, use_representative_path=False)
         ) == sorted(["project/example.py", "com/project/example.java"])
 
-    def test_strip_target(self) -> None:
+    def test_strip_sources_field(self) -> None:
+        source_root = "src/python/project"
+
+        def get_stripped_files_for_sources_field(
+            *,
+            source_files: Optional[List[str]],
+            sources_field_cls: Type[SourcesField] = SourcesField,
+            specified_source_files: Optional[List[str]] = None,
+        ) -> List[str]:
+            if source_files:
+                self.create_files(path=source_root, files=source_files)
+            sources_field = sources_field_cls(
+                source_files, address=Address.parse(f"{source_root}:lib")
+            )
```

Comment:
The ease of testing is something I'm really enjoying about Typed Fields :) compare this to the below of this file where we have to mock an entire `TargetAdaptor`.